fix: use interpolated address when performing health checks

MattJustMatt opened the pull request:
Fixes #18574

jrasell reviewed (changes requested):
Hi @MattJustMatt and thanks for the PR. The approach seems right to me, but. the change in function signature has broken some tests, which will need to be fixed up. It would also be nice to add some additional tests to cover the new behaviour and ensure interpolated service checks work as we now expected. Finally, a changelog entry for the addition can be included by running the `make cl` command.


Thanks again!

MattJustMatt commented:
@jrasell Thanks for the review! 

I've fixed the broken tests and added the changelog entry. I don't have a good enough understanding of the test infra to easily add a regression test for this, but if it's needed for this PR please let me know and I can dive in deeper.

shoenig commented on .changelog/18584.txt:
````diff
@@ -0,0 +1,3 @@
+```release-note:bug
+fix: use interpolated address when performing health checks
````
```suggestion
services: use interpolated address when performing nomad service health checks
```

MattJustMatt commented:
100% cool with me!
